client/packages/requisitions/src/ResponseRequisition/DetailView/IndicatorEdit/IndicatorLineEdit.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { Footer } from './Footer';
+import {
+  BasicTextInput,
+  Box,
+  IndicatorValueTypeNode,
+  InputWithLabelRow,
+  NumericTextInput,
+  useNotification,
+  useTranslation,
+} from '@openmsupply-client/common';
+import {
+  IndicatorLineRowFragment,
+  IndicatorLineWithColumnsFragment,
+  useResponse,
+} from '../../api';
+
+interface IndicatorLineEditProps {
+  requisitionNumber: number;
+  indicatorCode?: string;
+  hasNext: boolean;
+  next: IndicatorLineRowFragment | null;
+  hasPrevious: boolean;
+  previous: IndicatorLineRowFragment | null;
+  currentLine?: IndicatorLineWithColumnsFragment | null;
+}
+
+const INPUT_WIDTH = 185;
+const LABEL_WIDTH = '150px';
+
+export const IndicatorLineEdit = ({
+  requisitionNumber,
+  hasNext,
+  next,
+  hasPrevious,
+  previous,
+  currentLine,
+}: IndicatorLineEditProps) => {
+  const columns = currentLine?.columns.sort(
+    (a, b) => a.columnNumber - b.columnNumber
+  );
+  const t = useTranslation();
+  const { mutateAsync } = useResponse.document.updateIndicatorValue();
```

Comment:
Can you create a carryover for turning saving the indicator values in a draft before upserting. The values will then be saved on onBlur instead of onChange.

Saving it directly could cause data discrepancies if there are multiple indicators and the user enters the data quickly/navigates away quickly after entering data

Reply:
Don't think this can wait for carry over - at a minimum, we need to debounce the API call and have some frontend state... I tried with throttled network and everything I typed would get overwritten 😭